DAQ_FW/lib/IADCSensor/IADCSensor.cpp

```diff
@@ -18,67 +18,51 @@
 // CLASS METHODS:
 
 // Default constructor with generic metadata
-IADCSensor::IADCSensor(const char *_SensorName, uint16_t _SensorID, uint16_t _ADC_ID, uint16_t _ADC_GAIN_MODE)
-    : mSensorName(_SensorName), mSensorID(_SensorID), mADC_ID(_ADC_ID)
-{
-    initializeVoltagePerBit(_ADC_GAIN_MODE);
+IADCSensor::IADCSensor(const char *_SensorName, uint16_t _SensorID, uint16_t _ADC_ID)
+    : mSensorName(_SensorName), mSensorID(_SensorID), mADC_ID(_ADC_ID) {}
 
-    // obj declared when logger not active, could be hard to log
-    Logger::Notice("Class created with gain mode: %u and V/per bit of %D", _ADC_GAIN_MODE, mVOLTAGE_PER_BIT);
+// ADS1115 initialize func, to be called on setup()
+void IADCSensor::Initialize()
+{
+    if (!mADS.begin(ADS1X15_ADDRESS))
+    {
+        Logger::Error("Failed to start ads");
+        while (1)
```

Comment:
I've spent too many hours debugging error handling code like this on the VCU/BMS...

If we add other threads to the mix, it won't become absolutely clear that this thread is hanging. Especially since this is initialization code and the program likely hasn't done or logged anything important yet, let's just spam the log message like this

```
while(1)
{
    Logger::Error(msg)
}

Reply:
Also more information in the error message please! Let's include all of the useful metadata we added to our class in the error message.